helm/cluster-openstack/templates/_kubeadm_control_plane_template.tpl

```diff
@@ -14,6 +14,10 @@ spec:
           apiServer:
             extraArgs:
               cloud-provider: external
+              oidc-issuer-url: https://dex.{{ .Values.baseDomain }}
```

Comment:
It feels kind of weird requiring this to be passed in, but I don't see an easy alternative. If we could access values from `config` then we could default this based on the installation.

Reply:
In the dex app, when it is configured to be deployed in workload cluster we take [the domain from cluster values](https://github.com/giantswarm/dex-app/blob/master/helm/dex-app/templates/dex/secret.yaml#L33). So cluster operator (I think @rossf7 mentioned to me now cluster apps operator would do it in the new CAPI ecosystem) maintain cluster configmap values with
```
> » k get cm int1d-cluster-values -oyaml                                                                                     (giantswarm-flamingo/int1d)
apiVersion: v1
data:
  values: |
    baseDomain: int1d.k8s.flamingo.eu-west-1.aws.gigantic.io
    cluster:
      calico:
        CIDR: 10.237.0.0/16
      kubernetes:
        API:
          clusterIPRange: 10.238.0.0/16
        DNS:
          IP: 10.238.0.10
    clusterCA: |-
      -----BEGIN CERTIFICATE-----
      MIID...
```
can we rely on this on CAPO?

Reply:
There's a chicken and egg problem where the cluster-values config won't be generated until the helm release is created and the helm release won't be created until the config exists. The current idea is to add functionality to organization-operator to create a ConfigMap with installation-specific and organization-specific values such as `baseDomain`. Then we can merge those values into this app. We'll continue on this next week.